java/challenges/lib/src/main/java/challenges/stacksandqueues/PseudoQueue.java

```diff
@@ -0,0 +1,35 @@
+package challenges.stacksandqueues;
+
+import java.util.NoSuchElementException;
+
+public class PseudoQueue {
+  public Node top;
+  Stack stackOne;
+  Stack stackTwo;
+
+  public PseudoQueue(){
+    this.top = null;
+    stackOne = new Stack();
+    stackTwo = new Stack();
+  }
+
+  public void pseudoEnqueue(Object val){
+      stackOne.push(val);
+    }
+
+  public Object pseudoDequeue(){
+    Object val;
+    if (stackTwo.top == null){
```

Comment:
instead of running a double if(statement), you can wrap them both in one by using if(thing1 && thing2)